src/lib.rs

```diff
@@ -231,6 +233,18 @@ impl<T: Send> Input<T> {
         back_info & BACK_DIRTY_BIT == 0
     }
 
+    /// Access the input buffer directly, in non-mutable way
+    ///
+    /// This is simply a non-mutable version of `input_buffer()`.
+    /// For details, see the `input_buffer()` method.
+    ///
+    #[doc(hidden)]
+    fn peek_input_buffer(&self) -> &T {
+        // Access the input buffer directly
+        let input_ptr = self.shared.buffers[self.input_idx as usize].get();
+        unsafe { &*input_ptr }
+    }
```

Comment:
I don't think `#[doc(hidden)]` is needed here because this is not a `pub fn`, which means that it is already invisible in the docs.

FWIW, I agree that it's better to have symmetrical accessors on the input and output side. But you're right that it's better not to make a new public accessor called `peek`, when [the long-term goal is to rename these](https://github.com/HadrienG2/triple-buffer/issues/30).